dashboard/views/_committee.py

```diff
@@ -98,51 +96,44 @@ def committee_event(request, event_id):
         'form': form,
     }
 
-    return render(request, "committee_event.html", context)
+    return render(request, "committee-event.html", context)
 
 
-@verify_position(['Public Relations Chair', 'Scholarship Chair', 'Service Chair', 'Philanthropy Chair', 'Alumni Relations Chair', 'Membership Development Chair', 'Social Chair', 'Vice President of Health and Safety', ' Recruitment Chair', 'Vice President', 'President', 'Adviser'])
-def committee_event_add(request, position):
+@verify_position(['pr-chair', 'scholarship-chair', 'service-chair', 'philanthropy-chair', 'alumni-relations-chair', 'memdev-chair', 'social-chair', 'vphs', 'recruitment-chair', 'vice-president', 'president', 'adviser'])
+def committee_event_add(request, position_slug):
     """ Renders the committee meeting add page """
     form = CommitteeMeetingForm(request.POST or None)
+    position = Position.objects.get(title=position_slug)
 
     if request.method == 'POST':
         if form.is_valid():
             instance = form.save(commit=False)
-            committee = Position.objects.get(title=position).committee
+            committee = position.committee
             eligible_attendees = committee.members.order_by('last_name')
             instance.committee = committee
+            instance.name = committee.get_committee_display() + " Committee Meeting"
+            instance.slug = committee.chair.title
```

Comment:
isnt there a function that does this already? this code looks familiar from somewhere above

Reply:
This is done in the models as part of the init function for each of the child event classes. However, the issue is that for committee events, I want the name to be based on which committee the event is related to. As such, these values (name and slug) cannot be set in the init as technically these relationships (committee) doesn't exist yet at time of initialization.